server/src/main/java/org/elasticsearch/rest/CompatibleHandlers.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.xcontent.ToXContent;
+
+import java.util.function.Consumer;
+
+public class CompatibleHandlers {
+
+    /**
+     * TODO revisit when https://github.com/elastic/elasticsearch/issues/52370 is resolved
+     */
+    public static final String COMPATIBLE_HEADER = "Accept";
+    public static final String COMPATIBLE_PARAMS_KEY = "Compatible-With";
+    public static final String COMPATIBLE_VERSION = "7";
+
+    public static Consumer<RestRequest> consumeParameterType(DeprecationLogger deprecationLogger) {
```

Comment:
this method and the one below feel like REST v7 concerns. Can they be moved to the module ?

(which would leave this class pretty empty and possibly move the constant strings elsewhere ?)

Reply:
good point, the isV7Compatible can be made generic (applying to all compatible versions in the future) and can stay in RestController. 
How about the constants though? any ideas where to place them? It feels to me like `org.elasticsearch.rest` package would be good. Maybe RestUtils?

Reply:
`COMPATIBLE_HEADER` is only used in `RestRequest` (so there for now) (as a side note, we also need to also check `Content-Type`, but am good with that in a follow up)
`COMPATIBLE_VERSION` is only in `RestRequest` (so there for now)
`COMPATIBLE_PARAMS_KEY` is an x-content concept, so probably somewhere in x-content